raven-android/src/main/java/com/getsentry/raven/android/Util.java

```diff
@@ -0,0 +1,61 @@
+package com.getsentry.raven.android;
+
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+/**
+ * Raven Android utility methods.
+ */
+public final class Util {
+
+    /**
+     * Hide constructor.
+     */
+    private Util() {
+
+    }
+
+    /**
+     * Check whether the application has been granted a certain permission.
+     *
+     * @param ctx Android application context
+     * @param permission Permission as a string
+     * @return true if permissions is granted
+     */
+    public static boolean checkPermission(Context ctx, String permission) {
+        int res = ctx.checkCallingOrSelfPermission(permission);
+        return (res == PackageManager.PERMISSION_GRANTED);
+    }
+
+    /**
+     * Check whether the application has internet access at a point in time.
+     *
+     * @param ctx Android application context
+     * @return true if the application has internet access
+     */
+    public static boolean isConnected(Context ctx) {
```

Comment:
I don't make use of `isConnected` or `shouldAttemptToSend` right now... I'm not sure how this should communicate with `BufferedConnection`: https://github.com/getsentry/raven-java/blob/master/raven/src/main/java/com/getsentry/raven/connection/BufferedConnection.java#L88-L89

Maybe periodic runs are OK for now (or in general)? Or we can work some sort of connectivity notification system out inside the `Buffer` system.